pkg/aws/client.go

```diff
@@ -235,6 +236,46 @@ func (c *awsClient) ValidateCredentials() (bool, error) {
 	return true, nil
 }
 
+// Checks osdCcsAdmin IAM credentials are valid.
+// Get stack's key and actualy key and compare them
+// stack: aws cloudformation describe-stack-resource --logical-resource-id osdCcsAdminAccessKeys --stack-name osdCcsAdminIAMUser
+func (c *awsClient) ValidateCFUserCredentials() error {
+	name := AdminUserName
+	accessKeyInput := &iam.ListAccessKeysInput{
+		UserName: &name,
+	}
+	accessKeyList, err := c.iamClient.ListAccessKeys(accessKeyInput)
+	if err != nil {
+		return err
+	}
+
+	meta := accessKeyList.AccessKeyMetadata
+	keyList := make([]string, len(meta))
+	for _, key := range meta {
+		keyList = append(keyList, *key.AccessKeyId)
+	}
+
+	OsdCcsAdminStackNamePtr := OsdCcsAdminStackName
+	LogicalResourceIDPtr := "osdCcsAdminAccessKeys"
+	stackResourceInput := &cloudformation.DescribeStackResourceInput{
+		StackName:         &OsdCcsAdminStackNamePtr,
+		LogicalResourceId: &LogicalResourceIDPtr,
+	}
+	resources, err := c.cfClient.DescribeStackResource(stackResourceInput)
+	if err != nil {
+		return err
+	}
+	cfAccessKey := resources.StackResourceDetail.PhysicalResourceId
+
+	for _, item := range keyList {
+		if item == *cfAccessKey {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("credentials are not valid for %s \n you can recreate the CloudFormation stack with `moactl init --delete-stack`", name)
```

Comment:
The command to recreate would be something like `moactl init --delete-stack && moactl init`. Or maybe you can say something like "you can delete the CloudFormation stack with `moactl init --delete-stack` and create it again", or whatever, but I think the more important detail is the fact that if the user creates the CF stack in one region, then tries to read it from another, they will get an error. Not sure if there's a way to have a region-agnostic CF stack, or check all regions until we find the stack, or what - I'm still trying to figure out what to do there...

Reply:
I've changed the message body as requested. I think the permission related concerns are mostly related to OSD-4708 (I might be wrong but it currently exists in our backlog).